text/0000-reactive-assignments.md

```diff
@@ -666,6 +658,73 @@ It's not clear how this proposal affects standalone components (as opposed to th
 One consequence of the new way of doing things: Svelte no longer lives in `pkg.devDependencies`, but would instead need to become a `dependency`. Not very On Brand but probably a small price to pay.
 
 
+### Store
+
+A [store](https://svelte.technology/guide#state-management) can be attached to a component, and it will be passed down to all its children. This allows components to access (and manipulate) app-level data without needing to pass props around. Svelte 2 provides a convenient `$` syntax for referencing store properties:
+
+```html
+<h1>Hello {$name}!</h1>
+```
+
+Similarly, store methods can be invoked via event listeners:
+
+```html
+<button on:click="$set({ count: count + 1 })">increment</button>
+```
+
+The store can also be accessed programmatically:
+
+```js
+const { foo } = this.store.get();
+this.store.doSomething();
+```
+
+Since there is no longer a `this`, we need to reconsider this approach. At the same time, the `get`/`set`/`on`/`fire` interface (designed to mirror the component API) feels slightly anachronistic in light of the rest of this RFC. A major problem with the store in its current incarnation is its lack of typechecker-friendliness, and the undocumented hoops necessary to jump through to integrate with popular libraries like MobX.
+
+I'm not yet sure how the store fits into this proposal — this is perhaps the biggest unknown at this point.
```

Comment:
Not sure if I see the problem?

```html
<h1>Hello, {{name}}!</h1>

<script>
  import { ondestroy } from 'svelte';
  import { Store } from './my-store';
  
  // setup code
  export let name = 'world';
  const store = new Store({ name });

  const listener = store.on('state', ({ current }) => {
    // something changed... changes this component's "name"
    name = Math.random().toString(36).slice(7);
    // this component now updates
  });

  // cleanup
  ondestroy(listener.cancel);

Reply:
I think what that loses is being able to have the store Just Be There without having to manually import it in every component, with the different relative paths that will entail.

Perhaps we could use @TehShrike's `<svelte:meta bind:XXX/>` idea for this?

I'm not certain we want to maintain a magic way of referring to store properties in templates (the `$`-prefixed identifiers).

Reply:
Personally I'd like to get rid of magical store presence. Not everything _needs_ to connect to a/the store or have access to it.

Doing it this way also lets a develop to define a custom top-level store (or multiple) and import the ones they want within any given component.

As for `$vars` being accessible – I think that can still be achieved if we wanted to. If the compiler can link up `foobar` it should be able to link up `$foobar` too?

But, personally, I'd prefer to do away with magic `$vars` and have the store update the variables I've declared & wired up during setup. For example, the `export let name` being updated within store listener.

Reply:
> But, personally, I'd prefer to do away with magic $vars and have the store update the variables I've declared & wired up during setup. For example, the export let name being updated within store listener.

Strong disagree on this sentiment from me. We're using stores heavily and the ability for children to inherit parent stores and transparently reference `$vars` is super valuable to our sanity and iteration speed.

Reply:
Fair 😆 For context, I use store in less than 10% of my components.

Reply:
I'm with @tivac on this one.
The (relatively) simple & consistent (ie. same api) implementation of Store in v2 won a lot of people over imo - including me! By all means knock yourself out with immer etc  etc but in some parts of the real world it is ***very*** useful indeed